apps/ttalps_histogrammer.cpp

```diff
@@ -7,26 +7,33 @@
 #include "TTAlpsObjectsManager.hpp"
 #include "HistogramsFiller.hpp"
 #include "UserExtensionsHelpers.hpp"
+#include "ArgsManager.hpp"
 
 using namespace std;
 
-void CheckArgs(int argc, char **argv) {
-  if (argc != 2 && argc != 4 && argc != 6) {
-    fatal() << "Usage: " << argv[0] << " config_path"<<endl;
-    fatal() << "or"<<endl;
-    fatal() << argv[0] << " config_path input_path output_path"<<endl;
+int main(int argc, char **argv) {
+  
+  auto args = make_unique<ArgsManager>(argc, argv);
+  // check if optional value "config" is present
+  if (!args->GetString("config").has_value()) {
+    fatal() << "No config file provided" << endl;
     exit(1);
   }
```

Comment:
I was wondering if maybe we should have two sets of functions, something like`GetOptionalString` and `GetString`. The first one would work as it is now, but the second one would automatically print a fatal error and exit if the value was not found. Like this, we would avoid checking it every time on the "user's" side. If you think that makes sense, I can change that.